controllers/service.go

```diff
@@ -195,7 +196,7 @@ func (r *SingleClusterReconciler) createPodService(pName, pNamespace string) err
 		context.TODO(),
 		types.NamespacedName{Name: pName, Namespace: pNamespace}, service,
 	); err == nil {
-		return nil
+		return errors.NewAlreadyExists(schema.GroupResource{Resource: "Service"}, service.Name)
```

Comment:
In func createOrUpdatePodServiceIfNeeded, we need to rely on the error returned from here to check if service exist or not, otherwise had to do another client get call.